src/test/java/nextstep/subway/path/domain/PathFinderTest.java

```diff
@@ -0,0 +1,101 @@
+        아현 = new StationResponse(4L, "아현", LocalDateTime.now(), LocalDateTime.now());
+        충정로 = new StationResponse(5L, "충정로", LocalDateTime.now(), LocalDateTime.now());
+        서강대 = new StationResponse(6L, "서강대", LocalDateTime.now(), LocalDateTime.now());
+        공덕역 = new StationResponse(7L, "공덕역", LocalDateTime.now(), LocalDateTime.now());
+
+        // 노선
+        // 2호선
+        LineStationResponse 지하철2호선_홍대입구 = new LineStationResponse(홍대입구, null, 5, 5);
+        LineStationResponse 지하철2호선_신촌 = new LineStationResponse(신촌, 홍대입구.getId(), 5, 5);
+        LineStationResponse 지하철2호선_이대 = new LineStationResponse(이대, 신촌.getId(), 5, 5);
+        LineStationResponse 지하철2호선_아현 = new LineStationResponse(아현, 이대.getId(), 5, 5);
+        LineStationResponse 지하철2호선_충정로 = new LineStationResponse(충정로, 아현.getId(), 5, 5);
+
+        // 경의중앙선
+        LineStationResponse 경의중앙선_홍대입구 = new LineStationResponse(홍대입구, null, 7, 7);
+        LineStationResponse 경의중앙선_서강대 = new LineStationResponse(서강대, 홍대입구.getId(), 7, 7);
+        LineStationResponse 경의중앙선_공덕역 = new LineStationResponse(공덕역, 서강대.getId(), 7, 7);
+
+        // 공항철도
+        LineStationResponse 공항철도_홍대입구 = new LineStationResponse(홍대입구, null, 9, 9);
+        LineStationResponse 공항철도_공덕역 = new LineStationResponse(공덕역, 홍대입구.getId(), 9, 9);
+
+        // 지하철 5호선
+        LineStationResponse 지하철5호선_공덕역 = new LineStationResponse(공덕역, null, 10, 10);
+        LineStationResponse 지하철5호선_충정로 = new LineStationResponse(공덕역, 충정로.getId(), 10, 10);
+
+        // 노선 등록
+        지하철2호선 = new LineResponse(1L, "지하철2호선", "SKY_BLUE", LocalTime.now(), LocalTime.now(), 5, Arrays.asList(지하철2호선_홍대입구, 지하철2호선_신촌, 지하철2호선_이대, 지하철2호선_아현, 지하철2호선_충정로), LocalDateTime.now(), LocalDateTime.now());
+        경의중앙선 = new LineResponse(2L, "경의중앙선", "SKY_BLUE", LocalTime.now(),LocalTime.now(), 5, Arrays.asList(경의중앙선_홍대입구, 경의중앙선_서강대, 경의중앙선_공덕역), LocalDateTime.now(), LocalDateTime.now());
+        공항철도 = new LineResponse(3L, "공항철도", "SKY_BLUE", LocalTime.now(),LocalTime.now(), 5, Arrays.asList(공항철도_홍대입구, 공항철도_공덕역), LocalDateTime.now(), LocalDateTime.now());
+        지하철5호선 = new LineResponse(4L, "지하철5호선", "SKY_BLUE", LocalTime.now(),LocalTime.now(), 5, Arrays.asList(지하철5호선_공덕역, 지하철5호선_충정로), LocalDateTime.now(), LocalDateTime.now());
+
+        pathfinder = new PathFinder();
+        allLines = Arrays.asList(지하철2호선, 경의중앙선, 공항철도, 지하철5호선);
+    }
+
+    /**
+     * PathFinder 기능
+     * - 전체 노선도와 출발지, 도착지를 입력과 경로 탐색의 기준(거리, 시간)을 입력하면 최단 경로를 찾아준다
+     */
+    @DisplayName("전체 노선 중 입력받은 두 역간의 최단 경로를 찾는다")
+    @Test
+    void findShortestPathByDistance() {
+        // when
+        ShortestPathResult shortestPathResult = pathfinder.findShortestPath(allLines, 홍대입구.getId(), 충정로.getId(), PathFindType.DISTANCE);
+
+        // then
+        assertThat(shortestPathResult.getStations()).containsExactly(홍대입구, 공덕역, 충정로);
+        assertThat(shortestPathResult.getWeight()).isEqualTo(19);
+    }
+
+    @DisplayName("전체 노선 중 입력받은 두 역간의 최단 경로를 찾는다")
```

Comment:
최소 시간 경로를 찾는 테스트 맞죠?ㅎㅎ
```suggestion
    @DisplayName("전체 노선 중 입력받은 두 역간의 최소 시간 경로를 찾는다")
```